### PR TITLE
FS-1399 added intelligent ingestion (beta)

### PR DESCRIPTION
**Pull request by bkwi**
Added intelligent ingestion support.

Usage
```python
cli.upload(filepath='/path/to/your/image.jpg', intelligent=True)
```

**PLEASE DON'T MERGE YET**

### CODE REVIEW

File: filestack/utils/intelligent_ingestion.py [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+                    })
+
+                self._feed_uploaders()
+
+            elif response['worker'] == 'committer':
+                log.info('Got commit done message %s', response)
+                log.info('Removing part %s', response['part'])
+                self.parts.pop(response['part'])
+
+        if self._get_next_chunk()[1] is None:
```
Comment (nnkps) [inline filestack/utils/intelligent_ingestion.py]: why [1]?

File: filestack/utils/intelligent_ingestion.py [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+    while True:
+        job = commit_q.get(block=True)
+
+        if job == 'die':
+            break  # we need a way to stop it in tests (other than terminate())
+
+        log.info('Committer got job for part %s', job['part'])
+        log.debug('Job details: %s', job)
+
+        requests.post(
```
Comment (nnkps) [inline filestack/utils/intelligent_ingestion.py]: is there no response about commit result from api?

File: filestack/utils/intelligent_ingestion.py [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+            data={
+                'apikey': job['apikey'],
+                'uri': job['uri'],
+                'region': job['region'],
+                'upload_id': job['upload_id'],
+                'size': job['size'],
+                'part': job['part'],
+                'store_location': job['store_location']
+            },
+            files={'file': (job['filename'], '', None)},
```
Comment (bkwi) [inline filestack/utils/intelligent_ingestion.py]: when you specify `files` parameter, `requests` will send `multipart/form-data` POST, which is what we need here :) More info [here](http://docs.python-requests.org/en/master/user/quickstart/#post-a-multipart-encoded-file)

File: filestack/utils/intelligent_ingestion.py (hunk 2 above)
Comment (bkwi) [inline filestack/utils/intelligent_ingestion.py]: It doesn't return return anything that we need. Some error handling would be good, but I'll add it in a separate PR

File: filestack/utils/intelligent_ingestion.py (hunk 1 above)
Comment (bkwi) [inline filestack/utils/intelligent_ingestion.py]: this means there are no more chunks.

`_get_next_chunk()` returns a (part_num, chunk) tuple. It there are no more chunks, it returns (None, None). I'm just checking for chunk part ;)

**Pull request merged**